Mathlib/LinearAlgebra/TensorAlgebra/ToTensorPower.lean

```diff
@@ -118,18 +118,23 @@ theorem ofDirectSum_toDirectSum (x : TensorAlgebra R M) :
 #align tensor_algebra.of_direct_sum_to_direct_sum TensorAlgebra.ofDirectSum_toDirectSum
 
 @[simp]
+theorem mk_reindex_cast' {n m : ℕ} (h : Fin n = Fin m) (x : (⨂[R]^n) M) :
+    GradedMonoid.mk (A := fun i => (⨂[R]^i) M) m
+    (PiTensorProduct.reindex R M (Equiv.cast h) x) =
+    GradedMonoid.mk n x :=
+  Eq.symm (PiTensorProduct.gradedMonoid_eq_of_reindex_cast (fin_injective h) rfl)
+
```

Comment:
I think we should get rid of this, since type equalities are evil. Does the proof below break without it?

Reply:
For the same reason as above, the version with `h : n = m` is no longer a simp lemma. But removing it won't break things.

Reply:
I think all of the removed `simp` attributes should come with a comment referring to 7905; as far as I can tell, all of them look a bit iffy.

Reply:
This looks like a bug to me, https://github.com/leanprover/std4/issues/365.

I suggest you replace all of these with `simpNF`, and add a comment referencing `std4#365`.

Reply:
I've gone ahead and done that